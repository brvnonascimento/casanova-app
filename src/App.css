.App {
  display: grid;
  grid-template-columns: 10% 1fr 10%;
  grid-template-rows: 100px 0.2fr 0.1fr 0.2fr auto 0.2fr 100px;
  height: 100vh;
  overflow: hidden;
  box-sizing: border-box;
  justify-items: center;
}

header {
  grid-column-start: 2;
  place-self: center;
  /* which is the same as: */
  /* align-self: center;
  justify-self: center; */
}

/* main {
  grid-column-start: 2;
  grid-row-start: 2;
  grid-row-end: span 5;
  overflow-x: hidden;
  overflow-y: scroll;
} */

main {
  grid-row-start: 4;
  grid-column-start: 2;
  padding: 1rem;
}

form {
  margin-left: auto;
}



::webkit-scrollbar { 
  display: none; 
}