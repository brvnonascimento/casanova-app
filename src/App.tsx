import React, { useEffect, useReducer } from 'react';
import './App.css';
import { Form } from './components/main/form';

import { SheetResponse } from './services/api/googleSheets/types/response'
import { gapi } from './services/api/googleSheets'
import sheetsConfig from './config/apiKeys'
import {initialSheets, sheetReducer} from './reducers/sheetReducer'

const { apiKey, clientId, discoveryDocs, spreadsheetId } = sheetsConfig

const App: React.FC = () => {
  const [sheets, dispatch] = useReducer(sheetReducer, initialSheets)

  useEffect(() => {
    try {
      gapi.load("client", initClient)
    } catch (error) {
      console.log(error)
    }
  }, [])

  useEffect(() => {
    console.log(sheets)
  }, [sheets])

  const initClient = async () =>
    gapi.client
      .init({
        apiKey: apiKey,
        discoveryDocs: discoveryDocs,
        clientId: clientId,
        scope: 'https://www.googleapis.com/auth/spreadsheets.readonly'
      }).then(() => {
        gapi.client.load('sheets', 'v4', () => {
          gapi.client.sheets.spreadsheets.values
            .get({
              spreadsheetId: spreadsheetId,
              range: "A:K"
            })
            .then((res: SheetResponse) => {
              const data = res.result.values
              dispatch({
                type: 'NEW_SHEET',
                sheet: data
              })

              console.log(data)
              console.log(res)
            })
        })
      })



  return (
    <div className="App">
      <header>
        <h1>Casa Nova App</h1>
      </header>

      <main>
        {/* <Form /> */}
      </main>
    </div>
  );
}

export default App;
