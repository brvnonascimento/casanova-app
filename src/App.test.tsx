import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup)

it('should render without crashing', () => {
  const {asFragment} = render(<App />)

  expect(asFragment(<App />)).toMatchSnapshot()
})