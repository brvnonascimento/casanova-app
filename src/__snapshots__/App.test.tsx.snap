// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <h1>
        Casa Nova App
      </h1>
    </header>
    <main />
  </div>
</DocumentFragment>
`;
